function-aws-api-proxy/src/test/groovy/io/micronaut/function/aws/proxy/MicronautLambdaHandlerSpec.groovy

```diff
@@ -109,6 +110,95 @@ class MicronautLambdaHandlerSpec extends Specification {
         handler.close()
     }
 
+    @Issue("https://github.com/micronaut-projects/micronaut-aws/issues/1410")
+    void "POST form url encoded body binding to pojo works"() {
+        given:
+        MicronautLambdaContainerHandler handler = instantiateHandler()
+
+        when:
+        AwsProxyRequestBuilder builder = new AwsProxyRequestBuilder('/form', HttpMethod.POST.toString())
+        builder.header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_FORM_URLENCODED)
+        builder.body("message=World")
+        AwsProxyResponse response = handler.proxy(builder.build(), new MockLambdaContext())
+
+        then:
+        response.statusCode == 200
+        response.body == '{"message":"Hello World"}'
+    }
+
+    @PendingFeature
```

Comment:
this fails but works for Netty. we are not parsing any bodyArgument

Reply:
Trying to see why...

Reply:
Ahhh, I think it's here:

https://github.com/micronaut-projects/micronaut-aws/blob/619fae481b345ce4fe5e31e32bedb013649b9a76/function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/MicronautLambdaContainerHandler.java#L498-L500

And if we use `formUrlEncodedBodyToMap(body)`, we get a list of values for "message" from the conversion to map, which doesn't match the requested type